x-pack/plugins/security_solution/public/timelines/components/timeline/unified_components/data_table/index.tsx

```diff
@@ -0,0 +1,495 @@
+    const {
+      services: {
+        uiSettings,
+        fieldFormats,
+        dataViews,
+        storage,
+        dataViewFieldEditor,
+        notifications: { toasts: toastsService },
+        application: { capabilities },
+        theme,
+        data: dataPluginContract,
+      },
+    } = useKibana();
+
+    const [expandedDoc, setExpandedDoc] = useState<DataTableRecord & TimelineItem>();
+    const [fetchedPage, setFechedPage] = useState<number>(0);
+
+    const { browserFields, runtimeMappings } = useSourcererDataView(SourcererScopeName.timeline);
+
+    const showTimeCol = useMemo(() => !!dataView && !!dataView.timeFieldName, [dataView]);
+
+    const tableSettings = useMemo(() => {
+      const columnSettings = columns.reduce((acc, item) => {
+        if (item.initialWidth) {
+          acc[item.id] = { width: item.initialWidth };
+        }
+        return acc;
+      }, {} as Record<string, UnifiedDataTableSettingsColumn>);
+
+      return {
+        columns: columnSettings,
+      };
+    }, [columns]);
+
+    const defaultColumnIds = useMemo(() => columns.map((c) => c.id), [columns]);
+
+    const { timeline: { filterManager, rowHeight, sampleSize } = timelineDefaults } = useSelector(
+      (state: State) => timelineBodySelector(state, timelineId)
+    );
+
+    // const { activeStep, isTourShown, incrementStep } = useTourContext();
+
+    /* const isTourAnchor = useMemo(
+        () =>
+          isTourShown(SecurityStepId.alertsCases) &&
+          eventType === 'signal' &&
+          isDetectionsAlertsTable(timelineId) &&
+          ariaRowindex === 1,
+        [isTourShown, ariaRowindex, eventType, timelineId]
+      );
+      const onExpandEvent = useCallback(() => {
+        if (
+          isTourAnchor &&
+          activeStep === AlertsCasesTourSteps.expandEvent &&
+          isTourShown(SecurityStepId.alertsCases)
+        ) {
+          incrementStep(SecurityStepId.alertsCases);
+        }
+        onEventDetailsPanelOpened();
+      }, [activeStep, incrementStep, isTourAnchor, isTourShown, onEventDetailsPanelOpened]);*/
+
+    /* const onRowSelected: OnRowSelected = useCallback(
+        ({ eventIds, isSelected }: { eventIds: string[]; isSelected: boolean }) => {
+          dispatch(
+            timelineActions.setSelected({
+              id,
+              eventIds: getEventIdToDataMapping(data, eventIds, queryFields),
+              isSelected,
+              isSelectAllChecked:
+                isSelected && Object.keys(selectedEventIds).length + 1 === data.length,
+            })
+          );
+        },
+        [data, dispatch, id, queryFields, selectedEventIds]
+      );*/
+
+    const tableRows: Array<DataTableRecord & TimelineItem> = useMemo(
+      () =>
+        events.map(({ _id, _index, ecs, data }) => {
+          const _source = ecs as unknown as Record<string, unknown>;
+          const hit = { _id, _index: String(_index), _source };
+          return {
+            _id,
+            id: _id,
```

Comment:
....why 😂 . I know there's a reason and you're trying to match to an existing api..it's just annoying and unnecessary duplication and something else to track lol.